dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -237,6 +240,29 @@ private static void checkTokens(final Token quayToken, final Token githubToken,
         }
     }
 
+
+    @PUT
+    @Path("/{containerId}/insertDescriptorType")
+    @Timed
+    @UnitOfWork
+    @ApiOperation(value = "Endpoint needed to migrate descriptortype from a just a json property to being stored in the db",  hidden = true)
+    @Operation(operationId = "insertDescriptorTypes", description = "Endpoint needed to migrate descriptortype from a just a json property to being stored in the db", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    public Tool insertDescriptorType(@Parameter(hidden = true, name = "user") @Auth User user,
+            @Parameter(name = "containerId", description = "id of the tool", required = true, in = ParameterIn.PATH) @PathParam("containerId") Long containerId,
+            @Parameter(name = "body", description = "This is here to appease Swagger. It requires PUT methods to have a body, even if it is empty. Please leave it empty.") String emptyBody) {
```

Comment:
See https://en.wikipedia.org/wiki/Representational_state_transfer#Relationship_between_URI_and_HTTP_methods

Reply:
might want to elaborate

Reply:
I **think** this body could be written as a SQL statement, in which case, you don't need to introduce the endpoint and migrations would happen automatically.

Don't ask me to write it though. :)